src/app/assignrole/page.tsx

```diff
@@ -0,0 +1,92 @@
+// src/components/AssignRoleForm.tsx
+'use client';
+
+import React, { useState, FormEvent } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { assignRole } from '@/redux/slices/assignroleSlice';
+import { RootState } from '@/redux/store';
+import { useAppDispatch } from '@/redux/hooks/hook';
+import useToast from '@/components/alerts/Alerts';
+import { ToastContainer } from 'react-toastify';
+import { Button, ButtonStyle } from '@/components/formElements';
+import { AssignRoleFields, RoleKeys } from '@/utils/assingRoleFormField';
+import SideNav from '@/components/dashboard/SideNav';
+
+const AssignRoleForm: React.FC = () => {
+  const dispatch = useAppDispatch();
```

Comment:
we have already defined this dispatch which means you will get your function from dispatch